### DISCUSSION TITLE
Workflow seeding on Startup?

### DISCUSSION THREAD

**Discussion by faizu-619** [elsa-workflows/elsa-core#1582] [Q&A]
Is there anyway to create seeding process on startup application to generate default workflow?

**Comment by sfmskywalker** [comment]
Sure, there are multiple ways to generate a default workflow on application startup. Can you tell a little bit about your use case scenario?

**Comment by faizu-619** [comment]
Thanks for the reply @sfmskywalker 

In my case, I need to create or update default workflows for each Tenant once I deployed the application the seeding process will start and update the workflow from JSON file etc.

**Reply by sfmskywalker** [reply]
I see. And do you need these workflows to be editable by users once they are in the database?

If yes, then the easiest way would be to implement a startup task (create a class that implements IHostedService from where you read your JSON files, parse them and insert them into the database).

If no, then perhaps an even better solution would be to read the JSON workflows straight into Elsa by installing the `BlobStorageWorkflowProvider`.

That provider is like the `DatabaseWorkflowProvider` (which reads workflows form the configured database), but instead reads workflows from a configurable directory (which could be anywhere: local disk, Azure Blob Storage, FTP etc).

**Reply by chris1411** [reply]
We have this same use case. We store our json workflow files in the project with Elsa. Would love to see an example of how to add/update the workflows automatically on startup.

**Comment by Nokecy** [comment]
Does DatabaseWorkflowProvider override the definition in BlobStorageWorkflowProvider? @sfmskywalker

**Reply by sfmskywalker** [reply]
Nope, they work together. You can choose to register zero or many providers. Each one of them provides workflow blueprints to the system.

**Comment by faizu-619** [comment]
Hi @sfmskywalker, I'm implementing first option directly inserting on startup using dbContext. Thanks for the information

**Reply by sfmskywalker** [reply]
Awesome. Perhaps you could share a sample code snippet for the benefit of @chris1411 ?
